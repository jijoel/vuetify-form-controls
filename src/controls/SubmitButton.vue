<template>
<v-btn
  v-bind="$attrs"
  v-on="$listeners"
  :color="color"
  :loading="form.busy"
  :disabled="form.busy"
  type="submit"
>
  <slot>{{ label }}</slot>
</v-btn>
</template>

<script>
export default {

  name: 'SubmitButton',

  inheritAttrs: false,

  props: {
    color: {
      type: String,
      default: 'primary'
    },
    label: {
      type: String,
      default: 'Submit',
    },
    form: {
      type: Object,
      required: true,
    },
  }
}
</script>
