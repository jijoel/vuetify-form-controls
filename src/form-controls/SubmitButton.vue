<template>
<v-btn
  v-bind="$attrs"
  v-on="$listeners"
  :color="color"
  :loading="loading || form.busy"
  :disabled="disabled || form.busy || form.errors.any()"
  type="submit"
>
  <slot>{{ label }}</slot>
</v-btn>
</template>

<script>
export default {

  name: 'SubmitButton',

  inheritAttrs: false,

  props: {
    form: {
      type: Object,
      required: true,
    },
    color: {
      type: String,
      default: 'primary'
    },
    disabled: {
      type: Boolean,
      default: false,
    },
    label: {
      type: String,
      default: 'Submit',
    },
    loading: {
      type: Boolean,
      default: false,
    },
  },

}
</script>
