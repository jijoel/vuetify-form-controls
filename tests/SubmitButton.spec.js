import './setup'
import { shallowMount } from '@vue/test-utils'
import SubmitButton from '../src/controls/SubmitButton'

describe('SubmitButton', () => {

  test('it is rendered as custom vuetify button', () => {
    const wrapper = shallowMount(SubmitButton, {
        propsData: { form: jest.mock() }
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it has a slot for a label', () => {
    const wrapper = shallowMount(SubmitButton, {
        propsData: { form: jest.mock() },
        slots: { default: 'Do it' },
    })
    expect(wrapper.text()).toContain('Do it')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it can have label in prop', () => {
    const wrapper = shallowMount(SubmitButton, {
        propsData: { form: jest.mock(), label: 'Do it' },
    })
    expect(wrapper.text()).toContain('Do it')
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it can show loading status when form is busy', () => {
    const wrapper = shallowMount(SubmitButton, {
        propsData: { form: { busy: true }}
    })
    expect(wrapper.html()).toContain('loading')
    expect(wrapper.html()).toContain('disabled')
    expect(wrapper.element).toMatchSnapshot()
  })

})
