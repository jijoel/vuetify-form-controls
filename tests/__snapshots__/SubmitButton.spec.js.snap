// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton it can have label in prop 1`] = `
<v-btn-stub
  activeclass="v-btn--active"
  color="primary"
  tag="button"
  type="submit"
>
  Do it
</v-btn-stub>
`;

exports[`SubmitButton it can show loading status when form is busy 1`] = `
<v-btn-stub
  activeclass="v-btn--active"
  color="primary"
  disabled="true"
  loading="true"
  tag="button"
  type="submit"
>
  Submit
</v-btn-stub>
`;

exports[`SubmitButton it has a slot for a label 1`] = `
<v-btn-stub
  activeclass="v-btn--active"
  color="primary"
  tag="button"
  type="submit"
>
  Do it
</v-btn-stub>
`;

exports[`SubmitButton it is rendered as custom vuetify button 1`] = `
<v-btn-stub
  activeclass="v-btn--active"
  color="primary"
  tag="button"
  type="submit"
>
  Submit
</v-btn-stub>
`;
